api/rosetta/data_network_integration_test.go

```diff
@@ -0,0 +1,566 @@
+				},
+			},
+
+			checkError: checkRosettaError(
+				http.StatusUnprocessableEntity,
+				configuration.ErrorInvalidNetwork,
+				fmt.Sprintf("invalid network identifier network (have: %s, want: %s)", invalidNetwork, dps.FlowTestnet.String()),
+				map[string]interface{}{
+					"blockchain": dps.FlowBlockchain,
+					"network":    invalidNetwork,
+				},
+			),
+		},
+	}
+
+	for _, test := range tests {
+
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+
+			t.Parallel()
+
+			_, ctx, err := setupRecorder(statusEndpoint, test.request)
+			require.NoError(t, err)
+
+			err = api.Status(ctx)
+			test.checkError(t, err)
+		})
+	}
+}
+
+func TestAPI_Networks(t *testing.T) {
+
+	db := setupDB(t)
+	api := setupAPI(t, db)
+
+	// network request is basically an empty payload at the moment,
+	// there is a 'metadata' object that we're ignoring;
+	// but we can have the scaffolding here in case something changes
+
+	var netReq rosetta.NetworksRequest
+
+	rec, ctx, err := setupRecorder(listEndpoint, netReq)
+	require.NoError(t, err)
+
+	// execute the request
+	err = api.Networks(ctx)
+	assert.NoError(t, err)
+
+	var network rosetta.NetworksResponse
+	require.NoError(t, json.Unmarshal(rec.Body.Bytes(), &network))
+
+	if assert.Len(t, network.NetworkIDs, 1) {
+		assert.Equal(t, network.NetworkIDs[0].Blockchain, dps.FlowBlockchain)
+		assert.Equal(t, network.NetworkIDs[0].Network, dps.FlowTestnet.String())
+	}
+}
+
+func TestAPI_Options(t *testing.T) {
+
+	db := setupDB(t)
+	api := setupAPI(t, db)
+
+	const errorCount = 9
+
+	// verify version string is in the format of x.y.z
+	versionRe := regexp.MustCompile(`\d+\.\d+\.\d+`)
+
+	request := rosetta.OptionsRequest{
+		NetworkID: defaultNetwork(),
+	}
+
+	rec, ctx, err := setupRecorder(optionsEndpoint, request)
+
+	err = api.Options(ctx)
+	require.NoError(t, err)
+
+	// verify nominal case
+	assert.NoError(t, err)
+
+	assert.Equal(t, http.StatusOK, rec.Result().StatusCode)
+
+	var options rosetta.OptionsResponse
+	require.NoError(t, json.Unmarshal(rec.Body.Bytes(), &options))
+
+	// verify the version info
+	// TODO: we're not currently doing exact match of the version, only of the format
+	// check - should we?
+	assert.Regexp(t, versionRe, options.Version.RosettaVersion)
+	assert.Regexp(t, versionRe, options.Version.NodeVersion)
+	assert.Regexp(t, versionRe, options.Version.MiddlewareVersion)
+
+	assert.True(t, options.Allow.HistoricalBalanceLookup)
+
+	if assert.Len(t, options.Allow.OperationStatuses, 1) {
+
+		status := options.Allow.OperationStatuses[0]
+		assert.Equal(t, status.Status, dps.StatusCompleted)
+		assert.True(t, status.Successful)
+	}
+
+	if assert.Len(t, options.Allow.OperationTypes, 1) {
+		assert.Equal(t, options.Allow.OperationTypes[0], dps.OperationTransfer)
+	}
+
+	if assert.Len(t, options.Allow.Errors, errorCount) {
+
+		for i := uint(0); i < errorCount; i++ {
+
+			rosettaErr := options.Allow.Errors[i]
+
+			expectedCode := i + 1 // error codes start from 1
```

Comment:
Then why not start the loop at 1?

```go
for i := uint(1); i <= errorCount; i++ { ... }
```

Reply:
It felt a bit clearer to use the `i` as the loop variable for elements of `options.Allow.Errors`. Else I'd above use:
```
rosettaErr := options.Allow.Errors[i-1]
```

I'm ambivalent which is nicer😄

Reply:
I'm OK either way.